hoot-services/src/main/java/hoot/services/controllers/review/ReviewBookmarkResource.java

```diff
@@ -399,33 +398,40 @@ private static long remove(Long bookmarkId) {
      *
      * @param orderByCol
      *            - order by column to sort
-     * @param isAsc
-     *            - is order by asc | desc
      * @param limit
      *            - limit for numbers of returned results
      * @param offset
      *            - offset row for paging
      * @return - list of Review tags
      */
-    private static List<ReviewBookmarks> retrieveAll(String orderByCol, boolean isAsc, long limit,
-            long offset, Long[] creatorArray, Long[] layerArray) {
-        SQLQuery<ReviewBookmarks> query = createQuery().query().select(reviewBookmarks).from(reviewBookmarks);
+    private static SQLQuery<ReviewBookmarks> retrieveAll(long userId, String orderByCol, long limit, long offset, long creator, long layer) {
+        SQLQuery<ReviewBookmarks> query = createQuery()
+                .select(reviewBookmarks)
+                .from(reviewBookmarks)
+                .leftJoin(QMaps.maps).on(QMaps.maps.id.eq(reviewBookmarks.mapId));
 
-        if ((creatorArray != null) && (layerArray != null)) {
-            query.where(reviewBookmarks.createdBy.in((Number[]) creatorArray)
-                    .and(reviewBookmarks.mapId.in((Number[]) layerArray)));
+        if ((creator != -1) && (layer != -1)) {
+            query.where(reviewBookmarks.createdBy.eq(creator)
+                    .and(reviewBookmarks.mapId.eq(layer)));
         }
-        else if ((creatorArray != null) && (layerArray == null)) {
-            query.where(reviewBookmarks.createdBy.in((Number[]) creatorArray));
+        else if (creator != -1) {
+            query.where(reviewBookmarks.createdBy.eq(creator));
         }
-        else if ((creatorArray == null) && (layerArray != null)) {
-            query.where(reviewBookmarks.mapId.in((Number[]) layerArray));
+        else if (layer != -1) {
+            query.where(reviewBookmarks.mapId.eq(layer));
         }
         else {
             query.from(reviewBookmarks);
         }
 
-        query.orderBy(getSpecifier(orderByCol, isAsc));
+        // adds on to where clause a check to see if current user is able to see the maps associated with the bookmarks
+        if (!UserResource.adminUserCheck(DbUtils.getUser(userId))) {
+            query.where(
+                    QMaps.maps.userId.eq(userId).or(QMaps.maps.publicCol.isTrue())
```

Comment:
Need to use the `folders` and `foldermapmappings` join here.  The 'public' column on maps is unused for visibility (which is instead inherited from the containing folder).